### PR TITLE
Add support for new terrain data sources

### PR DESCRIPTION
**Pull request by stephendade**
This PR started as adding support for the SRTM1 dataset (https://terrain.ardupilot.org/SRTM1/) and allowing the user to select between the SRTM1 and SRTM3 datasets.

Along the way, I've made the ``terrain`` module the central place for all ground elevation requests where practical to do so. Previously, each module tended to have its own instance of the ``mp_elevation`` object.

The terrain data source can be switched at any time using ``terrain set source [SRTM1, SRTM3]`` or using the main menu on the map window.

The only outstanding issue is the ``misseditor`` GUI needs to be restarted if the terrain data source is changed. I've not quite worked out how to send updates to that GUI.

ping @auturgy

### CODE REVIEW

File: MAVProxy/modules/mavproxy_map/mp_elevation.py [hunk 1]
```diff
@@ -35,7 +40,7 @@ def GetElevation(self, latitude, longitude, timeout=0):
         '''Returns the altitude (m ASL) of a given lat/long pair, or None if unknown'''
         if latitude is None or longitude is None:
             return None
-        if self.database == 'srtm':
+        if self.database in ['SRTM1', 'SRTM3']:
```
Comment (tridge) [inline MAVProxy/modules/mavproxy_map/mp_elevation.py:43]: i am getting:
```
Traceback (most recent call last):
  File "/home/tridge/.local/lib/python3.10/site-packages/MAVProxy-1.8.52-py3.10.egg/MAVProxy/modules/mavproxy_map/mp_slipmap_ui.py", line 646, in on_mouse
  File "/home/tridge/.local/lib/python3.10/site-packages/MAVProxy-1.8.52-py3.10.egg/MAVProxy/modules/mavproxy_map/mp_slipmap_ui.py", line 475, in update_position
  File "/home/tridge/.local/lib/python3.10/site-packages/MAVProxy-1.8.52-py3.10.egg/MAVProxy/modules/mavproxy_map/mp_elevation.py", line 62, in GetElevation
UnboundLocalError: local variable 'alt' referenced before assignment
Migrating old SRTM folder
```

File: MAVProxy/modules/mavproxy_map/mp_elevation.py [hunk 2]
```diff
@@ -13,15 +13,20 @@
 
 from MAVProxy.modules.mavproxy_map import srtm
 
+TERRAIN_SERVICES = {
+    "SRTM1"      : ("terrain.ardupilot.org", "SRTM1"),
+    "SRTM3"      : ("terrain.ardupilot.org", "SRTM3")
```
Comment (tridge) [inline MAVProxy/modules/mavproxy_map/mp_elevation.py:18]: a comment saying that SRTM3 is 3 arcsecond, 90m, SRTM1 is 30m would be good

**Review by tridge — CHANGES_REQUESTED** [review]
need to test with no DISPLAY set, ensure that companion computer usage is OK

File: MAVProxy/modules/mavproxy_map/mp_elevation.py (hunk 2 above)
Comment (stephendade) [reply MAVProxy/modules/mavproxy_map/mp_elevation.py:20]: Add comments

**Comment by stephendade** [comment]
I've tested this on:
- Python 2.7
- Windows
- Headless Raspberry Pi

Needed to make a few minor changes, and works well with all of the above.

**Review by tridge — APPROVED** [review]

**Comment by tridge** [comment]
looks good, thanks!

**Pull request merged**